Nodejs/Product/Nodejs/Project/NodejsGeneralPropertyPageControl.cs

```diff
@@ -273,15 +255,21 @@ private void PortChanged(object sender, EventArgs e)
 
         private void WorkingDirChanged(object sender, EventArgs e)
         {
-            if (!this._workingDir.Text.Contains("$(") && !Directory.Exists(this._workingDir.Text))
-            {
-                this._nodeExeErrorProvider.SetError(this._workingDir, Resources.WorkingDirInvalidOrMissing);
-            }
-            else
+            var errorMessage = ValidateWorkingDir(this._workingDir.Text) ? "" : Resources.WorkingDirInvalidOrMissing;
+            this._nodeExeErrorProvider.SetError(this._workingDir, errorMessage);
+
+            Changed(sender, e);
+
+            bool ValidateWorkingDir(string workingDir)
             {
-                this._nodeExeErrorProvider.SetError(this._workingDir, string.Empty);
+                if (workingDir.Contains("$("))
+                {
+                    return true;
```

Comment:
This is because it's impossible to validate, rather than because it's valid, correct?

Reply:
Yes